python/ENTERPRISE_SUPPORT_PLAN_ENABLED/ENTERPRISE_SUPPORT_PLAN_ENABLED.py

```diff
@@ -0,0 +1,397 @@
+  Scenario: 3
+     Given: The AWS Account has access to "critical" case severity level as returned by the AWS Support API DescribeSeverityLevels
+      Then: Return COMPLIANT
+'''
+
+import json
+import sys
+import datetime
+import boto3
+import botocore
+from botocore.exceptions import ClientError
+
+try:
+    import liblogging
+except ImportError:
+    pass
+
+##############
+# Parameters #
+##############
+
+# Define the default resource to report to Config Rules
+DEFAULT_RESOURCE_TYPE = 'AWS::::Account'
+
+# Set to True to get the lambda to assume the Role attached on the Config Service (useful for cross-account).
+ASSUME_ROLE_MODE = False
+
+# Other parameters (no change needed)
+CONFIG_ROLE_TIMEOUT_SECONDS = 900
+
+#############
+# Main Code #
+#############
+
+# AWS Support API has only one endpoint which is in us-east-1 region
+def evaluate_compliance(event, configuration_item, valid_rule_parameters):
+    support_client = get_client('support', event, 'us-east-1')
+    try:
+        # Account with Basic or Developer support returns ClientError exception
+        response = support_client.describe_severity_levels()
+    except ClientError as error:
+        if error.response['Error']['Code'] == 'SubscriptionRequiredException':
+            return build_evaluation(
+                event['accountId'],
+                "NON_COMPLIANT",
+                event,
+                annotation="The AWS Enterprise Support Plan is not enabled for this AWS Account."
+            )
+        raise
+
+    for level in response['severityLevels']:
+        if level['code'] == 'critical':
+            return build_evaluation(event['accountId'], "COMPLIANT", event)
+
+    return build_evaluation(
+        event['accountId'],
+        "NON_COMPLIANT",
+        event,
+        annotation="The AWS Enterprise Support Plan is not enabled for this AWS Account."
+    )
+
+def evaluate_parameters(rule_parameters):
+    valid_rule_parameters = rule_parameters
+    return valid_rule_parameters
+
+####################
+# Helper Functions #
+####################
+
+# Build an error to be displayed in the logs when the parameter is invalid.
+def build_parameters_value_error_response(ex):
+    """Return an error dictionary when the evaluate_parameters() raises a ValueError.
+    Keyword arguments:
+    ex -- Exception text
+    """
+    return  build_error_response(internal_error_message="Parameter value is invalid",
+                                 internal_error_details="An ValueError was raised during the validation of the Parameter value",
+                                 customer_error_code="InvalidParameterValueException",
+                                 customer_error_message=str(ex))
+
+# This gets the client after assuming the Config service role
+# either in the same AWS account or cross-account.
+def get_client(service, event, region=None):
```

Comment:
Next time, please flag if you changed any functions in the template. It is really rare that it is needed and it should be considered together.

Reply:
I did highlight it: https://github.com/awslabs/aws-config-rules/pull/189#pullrequestreview-223588214
Note: Since AWS Support API has only one end-point (in us-east-1), I had to make changes in the boilerplate code to make this rule work. The changes are made in "get_client" and "get_assume_role_credentials" functions. If approved I will create a PR on the RDK repo as well to include this change.